samples/04.0-file-transfer/integration-tests/src/test/java/org/eclipse/dataspaceconnector/samples/FileTransferIntegrationTest.java

```diff
@@ -0,0 +1,240 @@
+import org.eclipse.dataspaceconnector.spi.types.domain.transfer.TransferProcessStates;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Map;
+import java.util.UUID;
+
+import static io.restassured.RestAssured.given;
+import static java.lang.String.format;
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static net.javacrumbs.jsonunit.assertj.JsonAssertions.assertThatJson;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.awaitility.Awaitility.await;
+import static org.eclipse.dataspaceconnector.common.configuration.ConfigurationFunctions.findUnallocatedServerPort;
+import static org.eclipse.dataspaceconnector.common.configuration.ConfigurationFunctions.propOrEnv;
+
+
+/**
+ * System Test for Sample 04.0-file-transfer
+ */
+@Tag("SystemTests")
+public class FileTransferIntegrationTest {
+
+    private static final String PROVIDER_ASSET_NAME = "test-document";
+
+    private static final String CONTRACT_NEGOTIATION_PATH = "/api/negotiation";
+    private static final String CONTRACT_AGREEMENT_PATH = "/api/control/negotiation/{contractNegotiationRequestId}";
+    private static final String TRANSFER_PATH = "/api/transfer/{transferId}";
+    private static final String FILE_TRANSFER_PATH = "/api/file/{filename}";
+
+    private static final String CONNECTOR_ADDRESS_PARAM = "connectorAddress";
+    private static final String CONTRACT_NEGOTIATION_REQUEST_ID_PARAM = "contractNegotiationRequestId";
+    private static final String TRANSFER_ID_PARAM = "transferId";
+    private static final String DESTINATION_PARAM = "destination";
+    private static final String CONTRACT_ID_PARAM = "contractId";
+    private static final String FILE_NAME_PARAM = "filename";
+
+    private static final String CONSUMER_ASSET_PATH = propOrEnv("edc.samples.04.consumer.asset.path", tempDirectory());
+    private static final int CONSUMER_CONNECTOR_PORT = findUnallocatedServerPort();
+    private static final String CONSUMER_CONNECTOR_HOST = "http://localhost:" + CONSUMER_CONNECTOR_PORT;
+
+    private static final int PROVIDER_CONNECTOR_PORT = findUnallocatedServerPort();
+    private static final String PROVIDER_CONNECTOR_HOST = "http://localhost:" + PROVIDER_CONNECTOR_PORT;
+    private static final String PROVIDER_ASSET_PATH = propOrEnv("edc.samples.04.asset.path", format("%s/%s.txt", tempDirectory(), PROVIDER_ASSET_NAME));
+
+    private static final String API_KEY_CONTROL_AUTH = propOrEnv("edc.api.control.auth.apikey.value", "password");
+    private static final String API_KEY_HEADER = "X-Api-Key";
+
+    @RegisterExtension
+    static EdcRuntimeExtension consumer = new EdcRuntimeExtension(
+            ":samples:04.0-file-transfer:consumer",
```

Comment:
Could be nice to use this name or a part of it to set a "runtime" variable inside the `ConsoleMonitor` to print the name of the runtime in the logs, this would help in test troubleshooting.
(there's already an unused constructor in `ConsoleMonitor` that accepts a `runtime` string and could be used for this need)

Reply:
Good idea, implemented. A prefix will now come up in the logs.


    [provider] DEBUG 2022-02-11T11:16:52.238847 [Provider] ContractNegotiation 8d0b11e1-15fb-4488-9df8-592d1cf13a07 is now in state CONFIRMING.
    [consumer] DEBUG 2022-02-11T11:16:52.261148 Response received from connector. Status 200